src/test/compile-fail/or-patterns.rs

```diff
@@ -0,0 +1,56 @@
+// should-ice
+#![allow(incomplete_features)]
+#![feature(or_patterns)]
+#![deny(unreachable_patterns)]
+
+// The ice will get removed once or-patterns are correctly implemented
+fn main() {
+    // We wrap patterns in a tuple because top-level or-patterns are special-cased for now.
+    match (0u8,) {
```

Comment:
Could we separate the test into 2?
- `exhaustiveness-unreachable-pattern.rs` (for "unreachable pattern")
- `exhaustiveness-is-exhaustive.rs` (for things that should pass, which is this one plus versions of the other tests below without the redundancy)

...and then a third test file for non-exhaustive matches.

Reply:
What would be the benefit of doing that ? I'd rather have similar-looking tests that exercise a same functionality grouped together

Reply:
You can still group the tests by the file names as I've done above. The benefit of splitting the tests is:
- We rely less on the correct functioning of compiletest and whether the compiler isn't emitting some errors due to phase order. A different tool or an alternate Rust implementation would merely have to check for `// {check,build,run}-pass` to see if this a test that needs to pass.
- We have a cleaner divide between "these things are in the language" and "these things are not" so the tests act as better specifications than if you mix failing and passing stuff together. I find this particularly important when stabilizing a feature from a language team perspective.